examples/simple-lambda.hs

```diff
@@ -7,26 +7,31 @@ import Control.Lens
 import qualified Data.ByteString.Lazy.Char8 as B
 import Data.Text (Text)
 import Stratosphere
+import Data.Aeson (object, Value(Array))
+import qualified Data.Vector as V
+import qualified Data.HashMap.Strict as HM
 
 main :: IO ()
 main = B.putStrLn $ encodeTemplate myTemplate
 
 myTemplate :: Template
 myTemplate =
   template
-  [ lambda ]
+  [ role, lambda ]
   & description ?~ "Lambda example"
   & formatVersion ?~ "2010-09-09"
 
 lambda :: Resource
 lambda =
-  resource "LambdaFunction" $
+  (resource "LambdaFunction" $
   LambdaFunctionProperties $
   lambdaFunction 
     lambdaCode 
     "index.handler" 
-    "arn:aws:iam::910653408535:role/MyLambdaBasicExecutionRole" 
+    "arn:aws:iam::910653408535:role/MyLambdaBasicExecutionRole"
```

Comment:
The only way I know of is to create this role inside CloudFormation, and then get the ARN either using `Ref` (depending on what type of resource it is), or possibly using [`Fn::GetAtt`](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html) with an arg of `Arn`. 

Luckily, the [`AWS::IAM::Role`](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html) resource allows you to use `Fn::GetAtt` to get the ARN!

```haskell
GetAtt "IAMRole" "Arn"
```